app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaViewModel.kt

```diff
@@ -0,0 +1,367 @@
+
+    private val searchQueryFlow = mutableSearchQueryFlow
+        .asStateFlow()
+        .debounce(SearchPeopleViewModel.DEFAULT_SEARCH_QUERY_DEBOUNCE)
+
+    private val _infoMessage = MutableSharedFlow<SnackBarMessage>()
+    val infoMessage = _infoMessage.asSharedFlow()
+
+    init {
+        viewModelScope.launch {
+            loadUserAvatar()
+            observeConversationWithSearch()
+        }
+    }
+
+    private fun loadUserAvatar() = viewModelScope.launch(dispatchers.io()) {
+        getSelf().collect { selfUser ->
+            withContext(dispatchers.main()) {
+                importMediaState = importMediaState.copy(avatarAsset = selfUser.previewPicture?.let {
+                    ImageAsset.UserAvatarAsset(wireSessionImageLoader, it)
+                })
+            }
+        }
+    }
+
+    private suspend fun observeConversationWithSearch() = viewModelScope.launch {
+        combine(
+            observeConversationListDetails()
+                .map {
+                    it.mapNotNull { conversationDetails ->
+                        conversationDetails.toConversationItem(
+                            wireSessionImageLoader,
+                            userTypeMapper
+                        )
+                    }
+                }, searchQueryFlow, selectedConversationFlow
+        ) { conversations, searchQuery, selectedConversations ->
+            val searchResult = if (searchQuery.isEmpty()) conversations else searchShareableConversation(
+                conversations,
+                searchQuery
+            )
+            ShareableConversationListState(
+                initialConversations = conversations,
+                searchQuery = searchQuery,
+                hasNoConversations = conversations.isEmpty(),
+                searchResult = searchResult,
+                conversationsAddedToGroup = selectedConversations.toImmutableList()
+            )
+        }
+            .flowOn(dispatchers.io())
+            .collect { updatedState ->
+                shareableConversationListState = updatedState
+            }
+    }
+
+    fun onSearchQueryChanged(searchQuery: TextFieldValue) {
+        val textQueryChanged = mutableSearchQueryFlow.value != searchQuery.text
+        // we set the state with a searchQuery, immediately to update the UI first
+        viewModelScope.launch {
+            if (textQueryChanged) {
+                mutableSearchQueryFlow.emit(searchQuery.text)
+            }
+        }
+    }
+
+    fun selectConversationOnRadioGroup(conversation: ConversationItem) = viewModelScope.launch {
+        selectedConversationFlow.emit(listOf(conversation))
+    }
+
+    fun onConversationClicked(conversationId: ConversationId) {
+        shareableConversationListState.initialConversations.find { it.conversationId == conversationId }?.let {
+            selectConversationOnRadioGroup(it)
+        }
+    }
+
+    @Suppress("LongMethod")
```

Comment:
can't we just use ConversationListViewModel in this screen so we won't duplicate the logic ?